### DISCUSSION TITLE
Add authorisation button to the doc

### DISCUSSION THREAD

**Discussion by szana8** [dedoc/scramble#53] [Ideas]
Hi,
As a standard API documentation it would be great to have an authorisation button and based on the auth type we can add the username and password, client_id and secret into a form and do the authentication than, use the token for the necessary request after. Like in swagger: https://swagger.io/docs/specification/authentication/ the ui look like in here: https://vivekcek.wordpress.com/2020/12/30/swaggerui-client-credentials-authentication-not-working-token_endpoint_auth_methodclient_secret_post-vs-client_secret_basic/
Without this we need postman or other service to get the bearer token than add it to the request in the ui.

Thanks,

**Comment by romalytvynenko** [comment]
Hey @szana8 

Please take a look at "Security" section in docs: https://scramble.dedoc.co/usage/security

It should do the job.

Thanks.

**Comment by szana8** [comment]
Yea, but if I choose clientCredentials I can not add my client id and secret in anywhere.

<img width="1433" alt="image" src="https://user-images.githubusercontent.com/4242050/194255233-f9eb5e09-e2ce-4b5d-bc06-0192130f8704.png">

It's empty.
This is my init code:
`Scramble::extendOpenApi(function (OpenApi $openApi) {
            $openApi->secure(
                SecurityScheme::oauth2()
                    ->flow('clientCredentials', function (OAuthFlow $flow) {
                        $flow
                            ->authorizationUrl('http://scrum-board.test/oauth/token')
                            ->addScope('place-orders', 'modify pets in your account');
                    })
            );
        });`

**Reply by romalytvynenko** [reply]
Hmm, I see. 

I'm wondering if this is something that Scramble generates incorrectly or Stoplight Elements 🤔

Would be cool if you can help me check this by inspecting generated document (`/docs/api.json`). Simply check if generated doc looks legit.

I'll also check it myself.

Thanks!

**Comment by szana8** [comment]
This is the api.json:
`{"openapi":"3.1.0","info":{"title":"laravel-scrum-board","version":"0.0.1"},"servers":[{"url":"http:\/\/localhost:8091\/api"}],"security":[{"oauth2":[]}],"paths":{"\/v1\/projects":{"get":{"summary":"Display a listing of the resource","tags":["Project"],"responses":{"200":{"description":"Array of `ProjectResource`","content":{"application\/json":{"schema":{"type":"array","items":{"$ref":"#\/components\/schemas\/ProjectResource"}}}}}}},"post":{"summary":"Store a newly created resource in storage","tags":["Project"],"requestBody":{"content":{"application\/json":{"schema":{"type":"object","properties":{"name":{"type":"string"},"description":{"type":"string"},"type":{"$ref":"#\/components\/schemas\/ProjectType"},"icon":{"type":"string"}},"required":["name","type"]}}}},"responses":{"200":{"description":"`ProjectResource`","content":{"application\/json":{"schema":{"$ref":"#\/components\/schemas\/ProjectResource"}}}}}}},"\/v1\/projects\/{project}":{"get":{"summary":"Display the specified resource","tags":["Project"],"parameters":[{"name":"project","in":"path","required":true,"description":"The project ID","schema":{"type":"integer"}}],"responses":{"200":{"description":"`ProjectResource`","content":{"application\/json":{"schema":{"$ref":"#\/components\/schemas\/ProjectResource"}}}}}},"put":{"summary":"Update the specified resource in storage","tags":["Project"],"parameters":[{"name":"project","in":"path","required":true,"description":"The project ID","schema":{"type":"integer"}}],"requestBody":{"content":{"application\/json":{"schema":{"type":"object","properties":{"name":{"type":"string"},"description":{"type":"string"},"type":{"$ref":"#\/components\/schemas\/ProjectType"},"icon":{"type":"string"}}}}}},"responses":{"200":{"description":"`ProjectResource`","content":{"application\/json":{"schema":{"$ref":"#\/components\/schemas\/ProjectResource"}}}}}},"delete":{"summary":"Remove the specified resource from storage","tags":["Project"],"parameters":[{"name":"project","in":"path","required":true,"description":"The project ID","schema":{"type":"integer"}}],"requestBody":{"content":{"application\/json":{"schema":{"type":"object"}}}},"responses":{"200":{"description":"","content":{"application\/json":{"schema":{"type":"boolean"}}}}}}}},"components":{"securitySchemes":{"oauth2":{"type":"oauth2","flows":{"clientCredentials":{"authorizationUrl":"http:\/\/scrum-board.test\/oauth\/token","scopes":{"place-orders":"modify pets in your account"}}}}},"schemas":{"ProjectResource":{"type":"object","properties":{"slug":{"type":"string"},"name":{"type":"string"},"type":{"type":"string"},"icon":{"type":["string","null"]},"owner":{"type":"object","properties":{"name":{"type":"string"},"email":{"type":"string"}},"required":["name","email"]}},"required":["slug","name","type","icon","owner"],"title":"ProjectResource"},"ProjectType":{"type":"string","enum":["SOFTWARE","SERVICE_DESK","BUSINESS"],"title":"ProjectType"}}}}`

And this is the bundled json:
`{
  "openapi": "3.1.0",
  "info": {
    "title": "laravel-scrum-board",
    "version": "0.0.1"
  },
  "servers": [
    {
      "url": "http://localhost:8091/api"
    }
  ],
  "security": [
    {
      "oauth2": []
    }
  ],
  "paths": {
    "/v1/projects": {
      "get": {
        "summary": "Display a listing of the resource",
        "tags": [
          "Project"
        ],
        "responses": {
          "200": {
            "description": "Array of `ProjectResource`",
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ProjectResource"
                  }
                }
              }
            }
          }
        }
      },
      "post": {
        "summary": "Store a newly created resource in storage",
        "tags": [
          "Project"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "object",
                "properties": {
                  "name": {
                    "type": "string"
                  },
                  "description": {
                    "type": "string"
                  },
                  "type": {
                    "$ref": "#/components/schemas/ProjectType"
                  },
                  "icon": {
                    "type": "string"
                  }
                },
                "required": [
                  "name",
                  "type"
                ]
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "`ProjectResource`",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProjectResource"
                }
              }
            }
          }
        }
      }
    },
    "/v1/projects/{project}": {
      "get": {
        "summary": "Display the specified resource",
        "tags": [
          "Project"
        ],
        "parameters": [
          {
            "name": "project",
            "in": "path",
            "required": true,
            "description": "The project ID",
            "schema": {
              "type": "integer"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "`ProjectResource`",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProjectResource"
                }
              }
            }
          }
        }
      },
      "put": {
        "summary": "Update the specified resource in storage",
        "tags": [
          "Project"
        ],
        "parameters": [
          {
            "name": "project",
            "in": "path",
            "required": true,
            "description": "The project ID",
            "schema": {
              "type": "integer"
            }
          }
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "object",
                "properties": {
                  "name": {
                    "type": "string"
                  },
                  "description": {
                    "type": "string"
                  },
                  "type": {
                    "$ref": "#/components/schemas/ProjectType"
                  },
                  "icon": {
                    "type": "string"
                  }
                }
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "`ProjectResource`",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProjectResource"
                }
              }
            }
          }
        }
      },
      "delete": {
        "summary": "Remove the specified resource from storage",
        "tags": [
          "Project"
        ],
        "parameters": [
          {
            "name": "project",
            "in": "path",
            "required": true,
            "description": "The project ID",
            "schema": {
              "type": "integer"
            }
          }
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "object"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "",
            "content": {
              "application/json": {
                "schema": {
                  "type": "boolean"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "securitySchemes": {
      "oauth2": {
        "type": "oauth2",
        "flows": {
          "clientCredentials": {
            "authorizationUrl": "http://scrum-board.test/oauth/token",
            "scopes": {
              "place-orders": "modify pets in your account"
            }
          }
        }
      }
    },
    "schemas": {
      "ProjectResource": {
        "type": "object",
        "properties": {
          "slug": {
            "type": "string"
          },
          "name": {
            "type": "string"
          },
          "type": {
            "type": "string"
          },
          "icon": {
            "type": [
              "string",
              "null"
            ]
          },
          "owner": {
            "type": "object",
            "properties": {
              "name": {
                "type": "string"
              },
              "email": {
                "type": "string"
              }
            },
            "required": [
              "name",
              "email"
            ]
          }
        },
        "required": [
          "slug",
          "name",
          "type",
          "icon",
          "owner"
        ],
        "title": "ProjectResource"
      },
      "ProjectType": {
        "type": "string",
        "enum": [
          "SOFTWARE",
          "SERVICE_DESK",
          "BUSINESS"
        ],
        "title": "ProjectType"
      }
    }
  }
}`

**Comment by romalytvynenko** [comment]
What is the bundled json? I don't understand the difference

**Comment by szana8** [comment]
I don't know I could generate a bundled json from the UI. I think that a formatted version or I don't know.

**Comment by romalytvynenko** [comment]
Hey @szana8 

I see. This looks like Stoplight Elements issue and generated docs are fine.

Let me see if I can do something with it.

**Comment by romalytvynenko** [comment]
Closing the issue as this is Spotlight Elements issue

**Comment by chrillep** [comment]
dont know if it makes any difference but here are the stoplight docs

https://docs.stoplight.io/docs/platform/37d160068e33c-export-files#reference-options
